python/requirements-dev.txt

```diff
@@ -0,0 +1,5 @@
+black==23.3.0
+mypy==1.3.0
+pylint==2.17.4
```

Comment:
I notice we're using pylint and ruff atm, for the record I'm all in favor of switching over to ruff-only

Reply:
Let's do it. I had left pylint because we hadn't discussed switching the github workflow. Could you update the workflow definition to lint with ruff, and drop pylint from requirements-dev.txt? Thanks!